codegen/arrai/svc_service.arrai

```diff
@@ -89,10 +89,11 @@ let sysl = //{./sysl};
                     }
                 }
             }
-
+            
             if result.HTTPResponse.StatusCode == http.StatusUnauthorized {
                 ${returnDownstreamError("Unauthorized", "result", "nil")}
             }
+            restlib.OnRestRequestResultHTTPResult(result, err, opts)
```

Comment:
This needs to go straight after `restlib.DoHTTPRequest` so that it always gets called regardless of whether it was an error or not.